src/Sylius/Bundle/CoreBundle/Doctrine/ORM/PaymentMethodRepository.php

```diff
@@ -26,4 +28,18 @@ public function createListQueryBuilder()
             ->leftJoin('o.translations', 'translation')
         ;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findEnabledForChannel(ChannelInterface $channel)
```

Comment:
Why won't we just use `$channel->getPaymentMethods()` collection? It implements `Doctrine\Common\Collections\Selectable`, so:

```php
$channel->getPaymentsMethods()->matching(new Criteria(new Comparison('enabled', Comparison::EQ, true))
```

Should work even faster (we won't need to hydrate all the payment methods only to pass them to another query).